website_event_crm/models/crm_lead.py

```diff
@@ -65,6 +48,30 @@ def _cron_auto_invite_website_event_type(self):
             except Exception:
                 _logger.exception("Failure trying to invite to website event type.")
 
+    @api.model
+    def _cron_auto_invite_website_event_type(self):
+        """Invite automatically leads/opportunities to website event categories."""
+        # Find opportunities
+        leads = self.search(
+            [
+                ("email_from", "!=", False),
+                ("event_type_id", "!=", False),
+                ("stage_id.auto_invite_website_event_type", "=", True),
+            ]
+        )
+        for company in [x.company_id for x in leads]:
```

Comment:
I guess that if you do it like this, you'll repeat companies. Try with:

```suggestion
        for company in leads.mapped("company_id"):
```

Reply:
I was considering leads with falsey `company_id` (the proper way would be `{x.company_id for x in leads}` but thinking twice, for this cases maybe we should avoid sending the invitation. And thinking again as I write this, maybe what I should consider in first place is that the company has a website where the user can view the notified events...